drivers/usb/uhc/uhc_mcux.c

```diff
@@ -0,0 +1,720 @@
+}
+
+/* Signal bus resume event, 20ms K-state + low-speed EOP */
+static int uhc_mcux_bus_resume(const struct device *dev)
+{
+	return uhc_mcux_bus_control(dev, kUSB_HostBusResume);
+}
+
+static void uhc_mcux_transfer_callback(void *param, usb_host_transfer_t *transfer,
+		usb_status_t status)
+{
+	const struct device *dev = (const struct device *)param;
+	struct uhc_transfer *const xfer = (struct uhc_transfer *const)(transfer->uhc_xfer);
+	struct uhc_data *data = dev->data;
+	struct uhc_mcux_data *priv = data->priv;
+	int err = -EIO;
+
+	if (status == kStatus_USB_Success) {
+		err = 0;
+
+		/* TODO: temp workaround because current stack doesn't support it. */
+		if ((xfer->ep == 0) && (transfer->setupPacket->bRequest == USB_SREQ_SET_ADDRESS)) {
+			priv->need_update_addr = true;
+		}
+	}
+
+#if defined(CONFIG_NOCACHE_MEMORY)
+	if (transfer->setupPacket != NULL) {
+		uhc_mcux_nocache_free(transfer->setupPacket);
+	}
+#endif
+	if ((xfer->buf != NULL) && (transfer->transferBuffer != NULL)) {
+		if (transfer->transferSofar > 0) {
+#if defined(CONFIG_NOCACHE_MEMORY)
+			memcpy(xfer->buf->__buf, transfer->transferBuffer, transfer->transferSofar);
+#endif
+			net_buf_add(xfer->buf, transfer->transferSofar);
+		}
+#if defined(CONFIG_NOCACHE_MEMORY)
+		uhc_mcux_nocache_free(transfer->transferBuffer);
+#endif
+	}
+
+	transfer->setupPacket = NULL;
+	transfer->transferBuffer = NULL;
+	k_mem_slab_free(&mcux_uhc_transfer_pool, transfer);
+	uhc_xfer_return(dev, xfer, err);
+}
+
+static int uhc_mcux_enqueue(const struct device *dev,
+				struct uhc_transfer *const xfer)
+{
+	uint8_t index;
+	usb_host_pipe_handle mcux_ep_handle = NULL;
+	const struct uhc_mcux_config *config = dev->config;
+	struct uhc_data *data = dev->data;
+	struct uhc_mcux_data *priv = data->priv;
+	usb_status_t status;
+	usb_host_transfer_t *mcux_xfer;
+	uint8_t ep;
+
+	uhc_xfer_append(dev, xfer);
+
+	/* firstly check and init the mcux endpoint handle */
+	for (index = 0; index < USB_HOST_CONFIG_MAX_PIPES; ++index) {
+		ep = xfer->ep;
+		if (ep == 0x80) {
+			ep = 0;
+		}
+
+		if ((priv->ep_handles[index].mcux_ep_handle != NULL) &&
+			(priv->ep_handles[index].ep == ep) &&
+			(priv->ep_handles[index].udev == xfer->udev)) {
+			mcux_ep_handle = priv->ep_handles[index].mcux_ep_handle;
+			break;
+		}
+	}
+
+	/* Initialize mcux endpoint pipe */
+	if (mcux_ep_handle == NULL) {
+		usb_host_pipe_init_t pipe_init;
+
+		for (index = 0; index < USB_HOST_CONFIG_MAX_PIPES; ++index) {
+			if (priv->ep_handles[index].mcux_ep_handle == NULL) {
+				break;
+			}
+		}
+
+		if (index >= USB_HOST_CONFIG_MAX_PIPES) {
+			return -ENOMEM;
+		}
+
+		pipe_init.devInstance = xfer->udev;
+		pipe_init.nakCount = USB_HOST_CONFIG_MAX_NAK;
+		pipe_init.maxPacketSize = xfer->mps;
+		pipe_init.endpointAddress = xfer->ep & 0x7Fu;
+		if (xfer->ep & 0x80u) {
+			pipe_init.direction = USB_IN;
+		} else {
+			pipe_init.direction = USB_OUT;
+		}
+		pipe_init.numberPerUframe = 0; /* TODO: need right way to implement it. */
```

Comment:
Could you add a broader comment here explaining:
- what we are working around here
- why this workaround is required

Is the issue that the UHC stack does not provide ways to set these parameters?